### ISSUE TITLE
SSL certificate problem: certificate has expired errors

### ISSUE THREAD

**Issue by bennyt2** [kreait/firebase-php#865]
### Describe the bug

I am getting a bunch of errors stating "cURL error 60: SSL certificate problem: certificate has expired". It's not on every request, but it's on some requests.

I've followed these steps: https://firebase-php.readthedocs.io/en/latest/troubleshooting.html#forbidden-errors, and it is still not working.

The "certificate has expired" makes me think it's a problem on the Firebase side. Would you agree, or could this still be some kind of issue on my server end?

### Installed packages

```json
abraham/twitteroauth                   6.2.0              The most popular PHP library for use with the Twitter OAuth REST API.
aws/aws-crt-php                        v1.2.4             AWS Common Runtime for PHP
aws/aws-sdk-php                        3.295.8            AWS SDK for PHP - Use Amazon Web Services in your PHP project
beberlei/doctrineextensions            v1.3.0             A set of extensions to Doctrine 2 that add support for additional query functions available in MySQL, Oracle, PostgreSQL and SQLite.
behat/transliterator                   v1.5.0             String transliterator
beste/clock                            3.0.0              A collection of Clock implementations
beste/json                             1.4.0              A simple JSON helper to decode and encode JSON
brick/math                             0.11.0             Arbitrary-precision arithmetic library
brightcove/api                         2.1                Brightcove PHP API Wrapper
composer/ca-bundle                     1.4.0              Lets you find a path to the system CA bundle, and includes a fallback to the Mozilla CA bundle.
composer/xdebug-handler                1.4.2              Restarts a process without Xdebug.
dms/phpunit-arraysubset-asserts        v0.5.0             This package provides ArraySubset and related asserts once deprecated in PHPUnit 8
doctrine/annotations                   1.14.3             Docblock Annotations Parser
doctrine/cache                         2.2.0              PHP Doctrine Cache library is a popular cache implementation that supports many different drivers such as redis, memcache, apc, mongodb and oth...
doctrine/collections                   1.8.0              PHP Doctrine Collections library that adds additional functionality on top of PHP arrays.
doctrine/common                        3.4.3              PHP Doctrine Common project is a library that provides additional functionality that other Doctrine projects depend on such as better reflectio...
doctrine/data-fixtures                 1.6.7              Data Fixtures for all Doctrine Object Managers
doctrine/dbal                          3.7.2              Powerful PHP database abstraction layer (DBAL) with many features for database schema introspection and management.
doctrine/deprecations                  1.1.2              A small layer on top of trigger_error(E_USER_DEPRECATED) or PSR-3 logging with options to disable all deprecations or selectively for packages.
doctrine/doctrine-bundle               2.11.1             Symfony DoctrineBundle
doctrine/doctrine-fixtures-bundle      3.4.5              Symfony DoctrineFixturesBundle
doctrine/doctrine-migrations-bundle    3.3.0              Symfony DoctrineMigrationsBundle
doctrine/event-manager                 1.2.0              The Doctrine Event Manager is a simple PHP event system that was built to be used with the various Doctrine projects.
doctrine/inflector                     2.0.8              PHP Doctrine Inflector is a small library that can perform string manipulations with regard to upper/lowercase and singular/plural forms of words.
doctrine/instantiator                  2.0.0              A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                         2.1.0              PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
doctrine/migrations                    3.5.5              PHP Doctrine Migrations project offer additional functionality on top of the database abstraction layer (DBAL) for versioning your database sch...
doctrine/orm                           2.17.2             Object-Relational-Mapper for PHP
doctrine/persistence                   2.5.7              The Doctrine Persistence project is a set of shared interfaces and functionality that the different Doctrine object mappers share.
doctrine/sql-formatter                 1.1.3              a PHP SQL highlighting library
egulias/email-validator                3.2.6              A library for validating emails against several RFCs
elastic/transport                      v8.1.0             HTTP transport PHP library for Elastic products
elasticsearch/elasticsearch            v8.2.2             PHP Client for Elasticsearch
emag-tech-labs/rabbitmq-bundle         2.5.0              Integrates php-amqplib with Symfony & RabbitMq. Formerly php-amqplib/rabbitmq-bundle, oldsound/rabbitmq-bundle.
escapestudios/symfony2-coding-standard 3.11.0             CodeSniffer ruleset for the Symfony 2+ coding standard
fakerphp/faker                         v1.23.1            Faker is a PHP library that generates fake data for you.
fig/http-message-util                  1.1.5              Utility classes and constants for use with PSR-7 (psr/http-message)
firebase/php-jwt                       v6.10.0            A simple library to encode and decode JSON Web Tokens (JWT) in PHP. Should conform to the current spec.
friendsofphp/proxy-manager-lts         v1.0.16            Adding support for a wider range of PHP versions to ocramius/proxy-manager
friendsofsymfony/jsrouting-bundle      2.8.0              A pretty nice way to expose your Symfony2 routing to client applications.
gedmo/doctrine-extensions              v3.13.0            Doctrine behavioral extensions
google/apiclient                       v2.15.3            Client library for Google APIs
google/apiclient-services              v0.331.0           Client library for Google APIs
google/auth                            v1.34.0            Google Auth Library for PHP
google/cloud-core                      v1.52.11           Google Cloud PHP shared dependency, providing functionality useful to all components.
google/cloud-firestore                 v1.14.0            Cloud Firestore client for PHP
google/cloud-storage                   v1.36.1            Cloud Storage Client for PHP
google/common-protos                   v4.5.0             Google API Common Protos for PHP
google/gax                             v1.26.2            Google API Core for PHP
google/grpc-gcp                        v0.3.0             gRPC GCP library for channel management
google/longrunning                     0.3.0              Google LongRunning Client for PHP
google/protobuf                        v3.25.2            proto library for PHP
grpc/grpc                              1.57.0             gRPC library for PHP
guzzlehttp/guzzle                      7.8.1              Guzzle is a PHP HTTP client library
guzzlehttp/oauth-subscriber            dev-master 8d6cab2 Guzzle OAuth 1.0 subscriber
guzzlehttp/promises                    2.0.2              Guzzle promises library
guzzlehttp/psr7                        2.6.2              PSR-7 message implementation that also provides common utility methods
j7mbo/twitter-api-php                  1.0.6              Simple PHP Wrapper for Twitter API v1.1 calls
kamranahmedse/php-geocode              dev-master 21d182d A wrapper around the Google Geocoding API to get different details such as latitude, longitude, country, city, district, postcode, town and str...
kevinpapst/adminlte-bundle             3.6.1              Admin theme bundle for Symfony 4+ based on AdminLTE 2.4 with FOSUserBundle support
knplabs/knp-snappy                     v1.4.4             PHP library allowing thumbnail, snapshot or PDF generation from a url or a html page. Wrapper for wkhtmltopdf/wkhtmltoimage.
knplabs/knp-snappy-bundle              v1.10.0            Easily create PDF and images in Symfony by converting Twig/HTML templates.
kreait/firebase-bundle                 5.1.1              Symfony Bundle for the Firebase Admin SDK
kreait/firebase-php                    7.9.1              Firebase Admin SDK
kreait/firebase-tokens                 4.3.0              A library to work with Firebase tokens
laminas/laminas-code                   4.13.0             Extensions to the PHP Reflection API, static code scanning, and code generation
lcobucci/clock                         3.2.0              Yet another clock abstraction
lcobucci/jwt                           5.2.0              A simple library to work with JSON Web Token and JSON Web Signature
league/csv                             9.5.0              Csv data manipulation made easy in PHP
league/fractal                         0.14.0             Handle the output of complex data structures ready for API output.
microsoft/azure-storage-blob           1.5.3              This project provides a set of PHP client libraries that make it easy to access Microsoft Azure Storage Blob APIs.
microsoft/azure-storage-common         1.5.2              This project provides a set of common code shared by Azure Storage Blob, Table, Queue and File PHP client libraries.
monolog/monolog                        2.9.2              Sends your logs to files, sockets, inboxes, databases and various web services
mtdowling/jmespath.php                 2.7.0              Declaratively specify how to extract elements from a JSON document
myclabs/deep-copy                      1.11.1             Create deep copies (clones) of your objects
nelmio/alice                           3.12.2             Expressive fixtures generator
nikic/php-parser                       v4.18.0            A PHP parser written in PHP
ninsuo/symfony-collection              2.1.33             A jQuery plugin that manages adding, deleting and moving elements from a Symfony collection
noweh/twitter-api-v2-php               3.4.0              This library provides methods for sending messages to Twitter and receiving statuses.
paragonie/constant_time_encoding       v2.6.3             Constant-time Implementations of RFC 4648 Encoding (Base-64, Base-32, Base-16)
paragonie/random_compat                v9.99.100          PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
pdepend/pdepend                        2.8.0              Official version of pdepend to be handled with Composer
phar-io/manifest                       2.0.3              Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                        3.2.1              Library for handling version information and constraints
php-amqplib/php-amqplib                v3.6.0             Formerly videlalvaro/php-amqplib.  This library is a pure PHP implementation of the AMQP protocol. It's been tested against RabbitMQ.
php-http/discovery                     1.14.2             Finds installed HTTPlug implementations and PSR-7 message factories
php-http/httplug                       2.3.0              HTTPlug, the HTTP client abstraction for PHP
php-http/promise                       1.1.0              Promise used for asynchronous HTTP requests
phpdocumentor/reflection-common        2.2.0              Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock      5.3.0              With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded in a DocBlock.
phpdocumentor/type-resolver            1.7.3              A PSR-5 based resolver of Class names, Types and Structural Element Names
phploc/phploc                          dev-main f07a8d6   A tool for quickly measuring the size of a PHP project
phpmd/phpmd                            2.8.2              PHPMD is a spin-off project of PHP Depend and aims to be a PHP equivalent of the well known Java tool PMD.
phpseclib/phpseclib                    3.0.35             PHP Secure Communications Library - Pure-PHP implementations of RSA, AES, SSH2, SFTP, X.509 etc.
phpstan/phpdoc-parser                  1.25.0             PHPDoc parser with support for nullable, intersection and generic types
phpunit/php-code-coverage              10.1.11            Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator              4.1.0              FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-invoker                    4.0.0              Invoke callables with a timeout
phpunit/php-text-template              3.0.1              Simple template engine.
phpunit/php-timer                      6.0.0              Utility class for timing
phpunit/phpunit                        10.5.5             The PHP Unit Testing framework.
predis/predis                          v2.2.2             A flexible and feature-complete Redis client for PHP.
psr/cache                              2.0.0              Common interface for caching libraries
psr/clock                              1.0.0              Common interface for reading the clock.
psr/container                          1.1.2              Common Container Interface (PHP FIG PSR-11)
psr/event-dispatcher                   1.0.0              Standard interfaces for event handling.
psr/http-client                        1.0.3              Common interface for HTTP clients
psr/http-factory                       1.0.2              Common interfaces for PSR-7 HTTP message factories
psr/http-message                       1.1                Common interface for HTTP messages
psr/link                               1.0.0              Common interfaces for HTTP links
psr/log                                1.1.4              Common interface for logging libraries
psr/simple-cache                       3.0.0              Common interfaces for simple caching
ralouphie/getallheaders                3.0.3              A polyfill for getallheaders.
ramsey/collection                      2.0.0              A PHP library for representing and manipulating collections.
ramsey/uuid                            4.7.5              A PHP library for generating and working with universally unique identifiers (UUIDs).
ramsey/uuid-doctrine                   2.0.0              Use ramsey/uuid as a Doctrine field type.
rize/uri-template                      0.3.5              PHP URI Template (RFC 6570) supports both expansion & extraction
rtheunissen/guzzle-log-middleware      v1.0.0             Guzzle middleware to log requests and responses
sebastian/cli-parser                   2.0.0              Library for parsing CLI options
sebastian/code-unit                    2.0.0              Collection of value objects that represent the PHP code units
sebastian/code-unit-reverse-lookup     3.0.0              Looks up which function or method a line of code belongs to
sebastian/comparator                   5.0.1              Provides the functionality to compare PHP values for equality
sebastian/complexity                   3.2.0              Library for calculating the complexity of PHP code units
sebastian/diff                         5.1.0              Diff implementation
sebastian/environment                  6.0.1              Provides functionality to handle HHVM/PHP environments
sebastian/exporter                     5.1.1              Provides the functionality to export PHP variables for visualization
sebastian/global-state                 6.0.1              Snapshotting of global state
sebastian/lines-of-code                2.0.2              Library for counting the lines of code in PHP source code
sebastian/object-enumerator            5.0.0              Traverses array structures and object graphs to enumerate all referenced objects
sebastian/object-reflector             3.0.0              Allows reflection of object attributes, including inherited and non-public ones
sebastian/phpcpd                       dev-main 7724245   Copy/Paste Detector (CPD) for PHP code.
sebastian/recursion-context            5.0.0              Provides functionality to recursively process PHP variables
sebastian/type                         4.0.0              Collection of value objects that represent the types of the PHP type system
sebastian/version                      4.0.1              Library that helps with managing the version number of Git-hosted PHP projects
sendgrid/php-http-client               3.1.0              HTTP REST client, simplified for PHP
sendgrid/sendgrid                      5.0.9              This library allows you to quickly and easily send emails through SendGrid using PHP.
sensio/framework-extra-bundle          v5.6.1             This bundle provides a way to configure your controllers with annotations
snc/redis-bundle                       4.7.0              A Redis bundle for Symfony
squizlabs/php_codesniffer              3.8.0              PHP_CodeSniffer tokenizes PHP, JavaScript and CSS files and detects violations of a defined set of coding standards.
stof/doctrine-extensions-bundle        v1.10.1            Integration of the gedmo/doctrine-extensions with Symfony
stripe/stripe-php                      v5.9.2             Stripe PHP Library
symfony/amqp-messenger                 v5.4.31            Symfony AMQP extension Messenger Bridge
symfony/asset                          v5.4.31            Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript files and image files
symfony/browser-kit                    v5.4.31            Simulates the behavior of a web browser, allowing you to make requests, click on links and submit forms programmatically
symfony/cache                          v5.4.34            Provides extended PSR-6, PSR-16 (and tags) implementations
symfony/cache-contracts                v2.5.2             Generic abstractions related to caching
symfony/config                         v5.4.31            Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                        v5.4.34            Eases the creation of beautiful and testable command line interfaces
symfony/css-selector                   v5.4.26            Converts CSS selectors to XPath expressions
symfony/debug-bundle                   v5.4.26            Provides a tight integration of the Symfony VarDumper component and the ServerLogCommand from MonologBridge into the Symfony full-stack framework
symfony/dependency-injection           v5.4.34            Allows you to standardize and centralize the way objects are constructed in your application
symfony/deprecation-contracts          v3.4.0             A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge                v5.4.34            Provides integration for Doctrine with various Symfony components
symfony/doctrine-messenger             v5.4.34            Symfony Doctrine Messenger Bridge
symfony/dom-crawler                    v5.4.32            Eases DOM navigation for HTML and XML documents
symfony/dotenv                         v5.4.34            Registers environment variables from a .env file
symfony/error-handler                  v6.3.5             Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher               v5.4.34            Provides tools that allow your application components to communicate with each other by dispatching events and listening to them
symfony/event-dispatcher-contracts     v3.4.0             Generic abstractions related to dispatching event
symfony/expression-language            v5.4.21            Provides an engine that can compile and evaluate expressions
symfony/filesystem                     v5.4.25            Provides basic utilities for the filesystem
symfony/finder                         v5.4.27            Finds files and directories via an intuitive fluent interface
symfony/flex                           v1.21.4            Composer plugin for Symfony
symfony/form                           v5.4.33            Allows to easily create, process and reuse HTML forms
symfony/framework-bundle               v5.4.34            Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-foundation                v5.4.34            Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                    v5.4.34            Provides a structured process for converting a Request into a Response
symfony/inflector                      v5.4.21            Converts words between their singular and plural forms (English only)
symfony/intl                           v5.4.30            Provides a PHP replacement layer for the C intl extension that includes additional data from the ICU library
symfony/lock                           v5.4.34            Creates and manages locks, a mechanism to provide exclusive access to a shared resource
symfony/mailer                         v5.4.34            Helps sending emails
symfony/messenger                      v5.4.31            Helps applications send and receive messages to/from other applications or via message queues
symfony/mime                           v5.4.26            Allows manipulating MIME messages
symfony/monolog-bridge                 v5.4.31            Provides integration for Monolog with various Symfony components
symfony/monolog-bundle                 v3.10.0            Symfony MonologBundle
symfony/options-resolver               v5.4.21            Provides an improved replacement for the array_replace PHP function
symfony/password-hasher                v6.4.0             Provides password hashing utilities
symfony/phpunit-bridge                 v5.4.34            Provides utilities for PHPUnit, especially user deprecation notices management
symfony/polyfill-ctype                 v1.28.0            Symfony polyfill for ctype functions
symfony/polyfill-intl-grapheme         v1.28.0            Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu              v1.28.0            Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn              v1.28.0            Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer       v1.28.0            Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring              v1.28.0            Symfony polyfill for the Mbstring extension
symfony/polyfill-php72                 v1.28.0            Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73                 v1.28.0            Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80                 v1.28.0            Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/polyfill-php81                 v1.28.0            Symfony polyfill backporting some PHP 8.1+ features to lower PHP versions
symfony/process                        v5.4.34            Executes commands in sub-processes
symfony/property-access                v5.4.26            Provides functions to read and write from/to an object or array using a simple string notation
symfony/property-info                  v5.4.24            Extracts information about PHP class' properties using metadata of popular sources
symfony/redis-messenger                v5.4.34            Symfony Redis extension Messenger Bridge
symfony/routing                        v5.4.34            Maps an HTTP request to a set of configuration variables
symfony/security-bundle                v5.4.34            Provides a tight integration of the Security component into the Symfony full-stack framework
symfony/security-core                  v5.4.30            Symfony Security Component - Core Library
symfony/security-csrf                  v5.4.27            Symfony Security Component - CSRF Library
symfony/security-guard                 v5.4.27            Symfony Security Component - Guard
symfony/security-http                  v5.4.31            Symfony Security Component - HTTP Integration
symfony/sendgrid-mailer                v5.4.23            Symfony Sendgrid Mailer Bridge
symfony/serializer                     v5.4.34            Handles serializing and deserializing data structures, including object graphs, into array structures or other formats like XML and JSON.
symfony/service-contracts              v2.5.2             Generic abstractions related to writing services
symfony/stopwatch                      v5.4.21            Provides a way to profile code
symfony/string                         v6.4.2             Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clusters in a unified way
symfony/templating                     v5.4.21            Provides all the tools needed to build any kind of template system
symfony/translation                    v5.4.31            Provides tools to internationalize your application
symfony/translation-contracts          v2.5.2             Generic abstractions related to translation
symfony/twig-bridge                    v5.4.34            Provides integration for Twig with various Symfony components
symfony/twig-bundle                    v5.4.31            Provides a tight integration of Twig into the Symfony full-stack framework
symfony/validator                      v5.4.34            Provides tools to validate values
symfony/var-dumper                     v5.4.29            Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter                   v5.4.32            Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-link                       v5.4.21            Manages links between resources
symfony/web-profiler-bundle            v5.4.34            Provides a development tool that gives detailed information about the execution of any request
symfony/workflow                       v5.4.34            Provides tools for managing a workflow or finite state machine
symfony/yaml                           v5.4.31            Loads and dumps YAML files
theseer/tokenizer                      1.2.2              A small library for converting tokenized PHP source code into XML and potentially other formats
twig/twig                              v2.16.0            Twig, the flexible, fast, and secure template language for PHP
twilio/sdk                             6.15.1             A PHP wrapper for Twilio's API
webmozart/assert                       1.11.0             Assertions to validate method input/output with nice error messages.
willdurand/jsonp-callback-validator    v1.1.0             JSONP callback validator.
```

### PHP version and extensions

```shell
composer               2.6.6    Composer package
composer-plugin-api    2.6.0    The Composer Plugin API
composer-runtime-api   2.2.2    The Composer Runtime API
ext-apcu               5.1.23   The apcu PHP extension
ext-bcmath             8.3.1    The bcmath PHP extension
ext-calendar           8.3.1    The calendar PHP extension
ext-ctype              8.3.1    The ctype PHP extension
ext-curl               8.3.1    The curl PHP extension
ext-date               8.3.1    The date PHP extension
ext-dom                20031129 The dom PHP extension
ext-exif               8.3.1    The exif PHP extension
ext-ffi                8.3.1    The FFI PHP extension
ext-fileinfo           8.3.1    The fileinfo PHP extension
ext-filter             8.3.1    The filter PHP extension
ext-ftp                8.3.1    The ftp PHP extension
ext-gd                 8.3.1    The gd PHP extension
ext-gettext            8.3.1    The gettext PHP extension
ext-grpc               1.60.0   The grpc PHP extension
ext-hash               8.3.1    The hash PHP extension
ext-iconv              8.3.1    The iconv PHP extension
ext-imap               8.3.1    The imap PHP extension
ext-json               8.3.1    The json PHP extension
ext-libxml             8.3.1    The libxml PHP extension
ext-mbstring           8.3.1    The mbstring PHP extension
ext-mysqli             8.3.1    The mysqli PHP extension
ext-mysqlnd            0        The mysqlnd PHP extension (actual version: mysqlnd 8.3.1)
ext-openssl            8.3.1    The openssl PHP extension
ext-pcntl              8.3.1    The pcntl PHP extension
ext-pcre               8.3.1    The pcre PHP extension
ext-pdo                8.3.1    The PDO PHP extension
ext-pdo_mysql          8.3.1    The pdo_mysql PHP extension
ext-pdo_sqlite         8.3.1    The pdo_sqlite PHP extension
ext-phar               8.3.1    The Phar PHP extension
ext-posix              8.3.1    The posix PHP extension
ext-protobuf           3.25.2   The protobuf PHP extension
ext-random             8.3.1    The random PHP extension
ext-readline           8.3.1    The readline PHP extension
ext-reflection         8.3.1    The Reflection PHP extension
ext-session            8.3.1    The session PHP extension
ext-shmop              8.3.1    The shmop PHP extension
ext-simplexml          8.3.1    The SimpleXML PHP extension
ext-sockets            8.3.1    The sockets PHP extension
ext-sodium             8.3.1    The sodium PHP extension
ext-spl                8.3.1    The SPL PHP extension
ext-sqlite3            8.3.1    The sqlite3 PHP extension
ext-sysvmsg            8.3.1    The sysvmsg PHP extension
ext-sysvsem            8.3.1    The sysvsem PHP extension
ext-sysvshm            8.3.1    The sysvshm PHP extension
ext-tokenizer          8.3.1    The tokenizer PHP extension
ext-xml                8.3.1    The xml PHP extension
ext-xmlreader          8.3.1    The xmlreader PHP extension
ext-xmlwriter          8.3.1    The xmlwriter PHP extension
ext-xsl                8.3.1    The xsl PHP extension
ext-zend-opcache       8.3.1    The Zend OPcache PHP extension
ext-zip                1.22.3   The zip PHP extension
ext-zlib               8.3.1    The zlib PHP extension
lib-curl               7.68.0   The curl library
lib-curl-libssh        0.9.3    curl libssh version
lib-curl-openssl       1.1.1.6  curl OpenSSL version (1.1.1.6)
lib-curl-zlib          1.2.11   curl zlib version
lib-date-timelib       2022.10  date timelib version
lib-date-zoneinfo      0        zoneinfo ("Olson") database for date
lib-fileinfo-libmagic  543      fileinfo libmagic version
lib-gd                 2.3.3    The gd library
lib-iconv              2.31     The iconv library
lib-libsodium          1.0.18   The libsodium library
lib-libxml             2.9.14   libxml library version
lib-libxslt            1.1.34   The libxslt library
lib-libxslt-libxml     2.9.10   libxml version libxslt is compiled against
lib-mbstring-libmbfl   1.3.2    mbstring libmbfl version
lib-mbstring-oniguruma 6.9.4    mbstring oniguruma version
lib-openssl            1.1.1.6  OpenSSL 1.1.1f  31 Mar 2020
lib-pcre               10.40    The pcre library
lib-pcre-unicode       14.0.0   PCRE Unicode version support
lib-pdo_sqlite-sqlite  3.31.1   The pdo_sqlite-sqlite library
lib-sqlite3-sqlite     3.31.1   The sqlite3-sqlite library
lib-zip-libzip         1.7.3    The zip-libzip library
lib-zlib               1.2.11   The zlib library
php                    8.3.1    The PHP interpreter
php-64bit              8.3.1    The PHP interpreter, 64bit
php-ipv6               8.3.1    The PHP interpreter, with IPv6 support
```

### Steps to reproduce the issue.

I'm not sure it's possible to reproduce outside of my server.

### Error message/Stack trace

```shell
I don't have an exact stack trace.

cURL error 60: SSL certificate problem: certificate has expired (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://trackmeet-io.firebaseio.com/meet_30391/event_detail/Individual-1099727.json

/srv/parser/vendor/kreait/firebase-php/src/Firebase/Exception/DatabaseApiExceptionConverter.php:58
message: Firebase PutRequestBuilder:execute() call failed
```

**Comment by jeromegamez** [comment]
Uh, that's a weird one! I checked the SSL certificate of your database URI and checked if I could retrieve the data (I could, are you sure the DB is supposed to be open?), and both turned out fine, at least on my end.

[I triggered the integration test suite](https://github.com/kreait/firebase-php/actions/runs/7977646666/job/21780978211) here on GitHub Actions - it has errors because of brittle tests, but no SSL certificate errors.

Local tests also didn't result in errors…

As much as I would like to help, at the moment it seems as if it was something on your server. What I would try is:

* Confirm that your server environment matches your local environment (assuming your outputs above are from your local computer)
* Update to PHP 8.3.2 (it would surprise me if that solved it, but, hey 🤷 😅)
* On the server, check the cURL version with `curl --version`, and update it, if possible (mine is 8.4.0)
* Check if there's a proxy in between (I don't know if that could even be a reason, I'm just brainstorming 😅)

* Try accessing the database directly with cURL (from your local machine and from your server), e.g. with
```
curl -iv -XGET https://trackmeet-io.firebaseio.com/meet_30391/event_detail/Individual-1099727.json
```
  This will also output detailed SSL information that could help with debugging

I hope this helps, at least a little!

**Comment by jeromegamez** [comment]
I just noticed you saying

>  It's not on every request, but it's on some requests.

That's another indication that it's not something on the Firebase side of things. Something seems to be off 🤔. I know it's dumb, but have you tried switching it off and on again? The server and/or the network? 🙈

**Comment by bennyt2** [comment]
Thanks so much for your feedback! Agree, it seems like something on my server. I'll give these things a try and post back here.

**Comment by njaw** [comment]
@bennyt2 I am also experiencing this issue. It started yesterday. It is also not for every request. I have opened a support ticket with Google and waiting for a response. I am in the process of re-releasing our application to production just to make sure it isn't an issue on the web servers themselves.

**Comment by jeromegamez** [comment]
Interesting! Thanks for chiming in, @njaw! I'm looking forward to your findings, this can't be a coincidence after all, and my suggestions have probably been useless 😅

**Comment by njaw** [comment]
@jeromegamez @bennyt2 Google support has reached out to me and mentioned that they have noted similar reports from other users and have escalated this to their Realtime Database team. I have provided them detailed logs of my issue. It seems this is something on their end as of now. My last occurrence was about 2 hours ago but I also redeployed our web application around that time so that it running on new Ubuntu instances. Whether that had any impact on the issue or not, I am not sure.
